### PR TITLE
Fix null hive-partition behavior in dask-cudf parquet

### PR DESCRIPTION
**Pull request by rjzamora**
## Description
This PR includes a few simple changes to fix the handling of null hive partitions in `dask_cudf`.
~Depends on https://github.com/dask/dask/pull/10007~

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — COMMENTED** [review]
Looks good, with some minor comments.

File: python/cudf/cudf/io/parquet.py [hunk 1]
```diff
@@ -1251,3 +1283,10 @@ def _default_open_file_options(
         )
     open_file_options["precache_options"] = precache_options
     return open_file_options
+
+
+def _hive_dirname(name, val):
+    # Simple utility to produce hive directory name
+    if pd.isna(val):
+        return f"{name}={HIVE_DEFAULT_PARTITION}"
+    return f"{name}={val}"
```
[the remark this answers is not in the document]
Comment (rjzamora) [reply python/cudf/cudf/io/parquet.py]: Thanks @bdice - Agree with your suggestions. The use of a constant comes from an earlier version of this change where the `HIVE_DEFAULT_PARTITION` was used in other places.

**Review by bdice — APPROVED** [review]
Looks good to me. I don't know a lot about hive partitioning to verify the test, but the code appears fine.

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**